packages/dd-trace/test/llmobs/tagger.spec.js

```diff
@@ -158,6 +158,34 @@ describe('tagger', () => {
 
         expect(Tagger.tagMap.get(span)).to.be.undefined
       })
+
+      it('uses the propagated mlApp over the global mlApp if both are provided', () => {
+        spanContext._trace.tags['_dd.p.llmobs_ml_app'] = 'my-propagated-ml-app'
+
+        tagger.registerLLMObsSpan(span, { kind: 'llm' })
+
+        const tags = Tagger.tagMap.get(span)
+        expect(tags['_ml_obs.meta.ml_app']).to.equal('my-propagated-ml-app')
+      })
+
+      describe('with no global mlApp configured', () => {
```

Comment:
nit: this is a little bit brittle as a test case since we're dependent on creating a new tagger - which could change in the future with refactorings. Ideally we could emulate the same initialization and configuration flow that a customer would but this would probably require a bigger test suite refactor so I'm ok to defer for now. Might be worth thinking about in the future.

Reply:
ah yeah we can make an integration test down the line, that's a good callout. 

we could actually do with more integration tests for llmobs here, where by integration test i mean we spin up a sandbox from a file. the only problem is those tests take forever to run, and wanted to try and see if we could test it in a unit test. but agree it is a bit brittle.

i also agree shared tests for this is a good follow up as well. i will tackle those and then come back to this repo to maybe backfill some tests locally 😄